api4/post.go

```diff
@@ -161,6 +167,9 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 
 		list, err = c.App.GetPostsBeforePost(c.Params.ChannelId, beforePost, c.Params.Page, c.Params.PerPage)
+		if len(list.Order) > 0 {
+			nextPostId = beforePost
+		}
 	} else {
 		etag = c.App.GetPostsEtag(c.Params.ChannelId)
```

Comment:
I think we can shortcut the next post id and previous post id here in a couple of cases:
1. If the page is 0, then the next post ID is empty
2. If `len(order) < c.Params.PerPage`, you're at the end of the channel, so the previous post ID is empty

Reply:
We can't apply this assumption since the base reference is also `afterPost` or `beforePost` which may be in the middle of posts order.

Reply:
Hmmm, I think it's applicable to `previousPostId` but not to `nextPostId`.  Let me check again.

Reply:
Only applied to `previousPostId` to prevent unnecessary call to `GetPreviousPostIdFromPostList`.